devito/ir/support/utils.py

```diff
@@ -79,7 +79,7 @@ def build_iterators(mapper):
     iterators = OrderedDict()
     for k, v in mapper.items():
         for d, offs in v.items():
-            if d.is_Stepping:
+            if d.is_Stepping and k is not None:
```

Comment:
Any code that uses a stepping dimension as either a plain symbol or an implicit_dimension generates this bug. Here is an example:

`from devito import Grid, TimeFunction, Eq, Operator
from numpy import int32

grid = Grid(shape=(50,), extent=(1.,))
u = TimeFunction(name = 'u', grid = grid)

# As an implicit dimension
u_eq = Eq(u.forward, u, implicit_dims = [grid.stepping_dim])

# As a plain symbol
#u_eq = Eq(u.forward, u + grid.stepping_dim)

op = Operator(u_eq)
`

Reply:
could you move this to a separate issue+PR+MFE+patch ?

also, the `if k is None: continue` could be added right below the `for k, v` loop

Reply:
This bug will be moved to a new PR once I am in Brazil.

Reply:
I'm not sure `grid.stepping_dim` should be used as an implicit dimension. Why would it be used over `grid.time_dim`? 

Also, I don't think it should be put into an equation either? I was going to suggest that `sympy.Mod` should be used instead, but that currently doesn't work and we have to write it as `grid.time_dim - c*floor(grid.time_dim/c)`.